src/Workspaces/CSharp/Portable/Extensions/ExpressionSyntaxExtensions.cs

```diff
@@ -799,6 +799,25 @@ private static bool CanReplace(ISymbol symbol)
                 }
             }
 
+            // Try to eliminate cases without actually calling CanReplaceWithReducedName. For expressions of the form
+            // 'this.Name' or 'base.Name', no additional check here is required.
+            if (!memberAccess.Expression.IsKind(SyntaxKind.ThisExpression, SyntaxKind.BaseExpression))
+            {
+                var actualSymbol = semanticModel.GetSymbolInfo(memberAccess.Name, cancellationToken);
+                if (actualSymbol.GetAnySymbol() is null)
+                {
+                    return false;
+                }
+
+                // Get a syntax node that's not part of the tree to make sure we speculate on just the name.
+                var speculativeTree = memberAccess.Name.WithoutTrivia();
```

Comment:
Oh, i see, you're removing trivia to get a node that's not in the tree.   Instead of this, could you just do SemanticModel.LookupSymbols with the name you have?